src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -92,113 +93,112 @@ public FilingsService(
+          throws SubmissionNotFoundException, ServiceException {
+
     var logMap = new HashMap<String, Object>();
     logMap.put(OVERSEAS_ENTITY_ID_KEY, overseasEntityId);
     logMap.put(TRANSACTION_ID_KEY, transaction.getId());
 
     Map<String, Object> userSubmission = new HashMap<>();
 
     OverseasEntitySubmissionDto submissionDto =
-        setSubmissionData(userSubmission, overseasEntityId, logMap);
-    setPaymentData(userSubmission, transaction, passThroughTokenHeader, logMap);
+            overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId)
+                    .orElseThrow(() -> new SubmissionNotFoundException(
+                            String.format("Empty submission returned when generating filing for %s", overseasEntityId)
+                    ));
 
     if (submissionDto.isForUpdate()) {
-      publicDataRetrievalService.initialisePublicData(
-          submissionDto.getEntityNumber(), passThroughTokenHeader);
-      privateDataRetrievalService.initialisePrivateData(submissionDto.getEntityNumber());
+      getPublicAndPrivateData(submissionDto.getEntityNumber(), passThroughTokenHeader);
+      setSubmissionDataForUpdates(userSubmission, submissionDto, logMap);
+      filing.setData(userSubmission);
+      filing.setKind(FILING_KIND_OVERSEAS_ENTITY_UPDATE);
+    } else {
+      setSubmissionData(userSubmission, submissionDto, logMap);
+      filing.setData(userSubmission);
+      filing.setKind(FILING_KIND_OVERSEAS_ENTITY);
     }
 
-    filing.setData(userSubmission);
     if (overseasEntitiesService.isSubmissionAnUpdate(requestId, overseasEntityId)) {
       filing.setCost(updateCostAmount);
     } else {
       filing.setCost(registerCostAmount);
     }
 
+    setPaymentData(userSubmission, transaction, passThroughTokenHeader, logMap);
     setDescriptionFields(filing, submissionDto.isForUpdate());
   }
 
-  private OverseasEntitySubmissionDto setSubmissionData(
-      Map<String, Object> data, String overseasEntityId, Map<String, Object> logMap)
-      throws SubmissionNotFoundException, ServiceException {
-    Optional<OverseasEntitySubmissionDto> submissionOpt =
-        overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId);
+  private void getPublicAndPrivateData(String entityNumber, String passThroughTokenHeader) throws ServiceException {
+    publicDataRetrievalService.initialisePublicData(entityNumber, passThroughTokenHeader);
+    privateDataRetrievalService.initialisePrivateData(entityNumber);
+  }
 
-    OverseasEntitySubmissionDto submissionDto =
-        submissionOpt.orElseThrow(
-            () ->
-                new SubmissionNotFoundException(
-                    String.format(
-                        "Empty submission returned when generating filing for %s",
-                        overseasEntityId)));
+  private void setSubmissionDataForUpdates(Map<String, Object> data, OverseasEntitySubmissionDto submissionDto, Map<String, Object> logMap) {
```

Comment:
This should also get a different arg or extra arg - the actual UpdateSubmission model. I also find the naming of this method confusing as is. Can we just replace where this is called by a code comment and remove this empty method. I've not checked SonarQube but would expect it to complain.

Reply:
Agreed. Comment added to replace method call.